Bottom: 74c905b922f8b0a1097ea6665a1e752a27bb9e9e
Top:    585026a4baecf1fdc9c3be4119abadd61d7e943b
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-13 15:41:32 -0500

Refresh of buildbot-fixup

---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index 166390e6..e019df23 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -46,6 +46,7 @@
     asset_host: "{{hostvars['nexus']['uri']}}"
     asset_user: "buildbot"
     asset_password: "{{nexus_buildbot_password}}"
+    buildbot_master: "{{hostvars[buildbot_hostname].ip_addr}}"
     container_etc_hosts:
       nexus.novatech-llc.com: "{{hostvars['nexus']['ip_addr']}}"
     known_hosts: "{{ gitlab_keys.stdout }}"
diff --git a/roles/buildbot-master/files/openembedded.py b/roles/buildbot-master/files/openembedded.py
index 7384ec25..a2a59480 100755
--- a/roles/buildbot-master/files/openembedded.py
+++ b/roles/buildbot-master/files/openembedded.py
@@ -197,15 +197,17 @@ def ComputeBuildProperties(props):
 
     newprops['timestamp'] = CurrentTime()
 
-    version = props.getProperty('version', default=newprops['timestamp'])
+    version = props.getProperty('version')
+    if not version:
+        version = newprops['timestamp']
 
-    urlpath = 'ntel/unofficial'
+    urlpath = 'unofficial'
 
-    if props.getProperty('release_pin') is True:
-        urlpath = 'ntel/release'
+    if props.getProperty('release_pin'):
+        urlpath = 'release'
 
     if props.getProperty('scheduler') == 'ntel-nightly':
-        urlpath = 'ntel/nightly'
+        urlpath = 'nightly'
 
     urlpath = os.path.join('ntel', urlpath, version)
 
@@ -346,7 +348,7 @@ class BitBakeArchive(steps.ShellSequence):
             artfile = os.path.join(art['dest'], art['artifact'])
             commands.append(util.ShellArg(
                 command=[
-                    'scripts/ci-archive.sh',
+                    'ci-archive.sh',
                     art['prefix'],
                     artfile
                 ],
diff --git a/roles/buildbot-master/files/ptxdist.py b/roles/buildbot-master/files/ptxdist.py
index c6e563b8..aab8d231 100644
--- a/roles/buildbot-master/files/ptxdist.py
+++ b/roles/buildbot-master/files/ptxdist.py
@@ -272,7 +272,9 @@ def ComputeBuildProperties(props):
 
     newprops['timestamp'] = CurrentTime()
 
-    version = props.getProperty('version', default=timestamp)
+    version = props.getProperty('version')
+    if not version:
+        version = newprops['timestamp']
 
     newprops['project'] = "OrionLX-%s-glibc" % (
         props.getProperty("platform")
@@ -348,7 +350,7 @@ def isReleaseBuild(step):
     return False
 
 # Create build factory for ptxdist
-ptxdist_factory = util.BuildFactory(
+ptxdist_factory = util.BuildFactory([
     steps.SetProperties(ComputeBuildProperties),
 
     steps.MakeDirectory(dir=util.Property('artifact_dest')),
@@ -369,7 +371,7 @@ ptxdist_factory = util.BuildFactory(
     PTXDistImages(
         doStepIf=isReleaseBuild,
     ),
-)
+])
 
 # ptxdist_factory.addStep(steps.ShellCommand(command=["./scripts/build-upgrade-test.sh"]))
 # ptxdist_factory.addStep(steps.ShellCommand(command=["curl", "--progress-bar", "-o", "/dev/null", "http://george:1234@172.16.190.70/outlet?1=CCL"]))
diff --git a/roles/buildbot-worker-ntel/defaults/main.yml b/roles/buildbot-worker-ntel/defaults/main.yml
index 0f440dc6..c6097fe5 100644
--- a/roles/buildbot-worker-ntel/defaults/main.yml
+++ b/roles/buildbot-worker-ntel/defaults/main.yml
@@ -54,6 +54,7 @@ container_volumes:
 - '{{ data_volume }}:{{ data_path }}:z'
 - '{{ cache_volume }}:{{ cache_path }}:z'
 - '{{ secrets_hostdir }}:{{ secrets_path }}'
+- '{{ config_hostdir }}/ci-archive.sh:/usr/local/bin/ci-archive.sh'
 - '{{ config_hostdir }}/pydistutils.cfg:/home/buildbot/.pydistutils.cfg'
 - '{{ config_hostdir }}/ptxdistrc:/home/buildbot/.ptxdist/ptxdistrc-2012.09'
 - '{{ config_hostdir }}/netrc:/home/buildbot/.netrc'
diff --git a/roles/buildbot-worker-ntel/files/ci-archive.sh b/roles/buildbot-worker-ntel/files/ci-archive.sh
new file mode 100644
index 00000000..57c3c260
--- /dev/null
+++ b/roles/buildbot-worker-ntel/files/ci-archive.sh
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -x
+PREFIX="$1";shift
+ARCHIVE="$1";shift
+
+mkdir -p "$(dirname $ARCHIVE)"
+
+# Gather variables from bitbake
+bitbake_vars="$(mktemp)"
+bitbake -e orion-headless-image 2>&1 > $bitbake_vars
+eval $(grep '^TOPDIR=' $bitbake_vars)
+eval $(grep '^TMPDIR=' $bitbake_vars)
+eval $(grep '^DEPLOY_DIR=' $bitbake_vars)
+eval $(grep '^BUILDHISTORY_DIR=' $bitbake_vars)
+
+# Gather files to archive into a text file to feed to tar
+files_to_archive="$(mktemp)"
+echo "${DEPLOY_DIR}" >> $files_to_archive
+echo "${BUILDHISTORY_DIR}" >> $files_to_archive
+find "${TMPDIR}" -type d -name "temp" >> $files_to_archive
+echo "$*" >> $files_to_archive
+
+# Remove leading and trailing slashes from TOPDIR
+TOPDIR=${TOPDIR#/}; TOPDIR=${TOPDIR%/}
+
+# Create the archive, substituting path prefixes of TOPDIR into rootdir
+tar --verbose --create --auto-compress \
+    --transform "s!^${TOPDIR}!${PREFIX}!" \
+    --file "${ARCHIVE}" \
+    --files-from="${files_to_archive}"
+
+# Clean up
+rm $files_to_archive
+rm $bitbake_vars
diff --git a/roles/buildbot-worker-ntel/tasks/main.yml b/roles/buildbot-worker-ntel/tasks/main.yml
index 7c865e5c..503c9c5e 100644
--- a/roles/buildbot-worker-ntel/tasks/main.yml
+++ b/roles/buildbot-worker-ntel/tasks/main.yml
@@ -61,6 +61,7 @@
     dest: '{{ config_hostdir }}/pydistutils.cfg'
     group: '{{ buildbot_uid }}'
     owner: '{{ buildbot_uid }}'
+    mode: "u=rw,g=,o="
 
 - name: Populate assets secret
   template:
@@ -70,6 +71,14 @@
     owner: '{{ buildbot_uid }}'
     mode: "u=rw,g=,o="
 
+- name: Populate archival script
+  copy:
+    src: ci-archive.sh
+    dest: '{{ config_hostdir }}/ci-archive.sh'
+    group: '{{ buildbot_uid }}'
+    owner: '{{ buildbot_uid }}'
+    mode: 'u=rwx,g=rx,o=rx'
+
 - name: Create secrets volume
   file:
     path: '{{ secrets_hostdir }}'
@@ -81,6 +90,7 @@
     dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
     group: '{{ buildbot_uid }}'
     owner: '{{ buildbot_uid }}'
+    mode: "u=rw,g=,o="
 
 - name: Checkout image repo
   git:
@@ -95,7 +105,7 @@
     tag: '{{ docker_image_tag }}'
     path: '{{ image_dir }}'
     buildargs: '{{ image_args }}'
-    extra_hosts: '{{ container_etc_hosts }}'
+    etc_hosts: '{{ container_etc_hosts }}'
     force: true
 
 - name: Start buildbot worker container
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index c3a63545..e4c3fccd 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -106,7 +106,7 @@
     tag: '{{ docker_image_tag }}'
     path: '{{ image_dir }}'
     buildargs: '{{ image_args }}'
-    extra_hosts: '{{ container_etc_hosts }}'
+    etc_hosts: '{{ container_etc_hosts }}'
     force: true
 
 - name: Start buildbot worker container
